### ISSUE TITLE
Implement "Like," "Book," and "Request Management" Features with Firebase Integration

### ISSUE THREAD

**Issue by Siddesh0002T** [RubbySoft/fixmate#4]
I need a developer to implement and fix functionality in my Next.js project using TypeScript, Tailwind, and Firebase. Below are the requirements and current issues:

### **1. Like Button on Profile Page**

- **Page:** http://localhost:3000/fine-professional
- **Feature:**
 **When a user clicks the "Like" button on a worker's profile:**
Increment the "like" count for that worker in Firebase.
Reflect the updated like count on the UI in real-time.

### **2. Book Button on Profile Page**

- **Page:** http://localhost:3000/fine-professional
- **Feature:**
 **When a logged-in user clicks the "Book" button:**
      Fetch the logged-in user's data (e.g., name, phone number, etc.) from Firebase.
      Create a booking request for the worker.
      Store the booking request in Firebase.

### **3. Dashboard for Bookings**

- **Page:** http://localhost:3000/dashboard/bookings
- **Feature:**
    **Display booking requests for workers.**
       Add an "Accept" button for workers to accept a booking request:
       Upon acceptance, set the booking status to "Pending" in Firebase.
       Add a "Complete" button for workers to mark the booking as completed:
       Upon clicking, update the booking status to "Completed" in Firebase.

### **4. Rate Button**
**Feature:**
      Implement a "Rate" button to allow users to rate the worker.
      Store the rating in Firebase and update the worker's profile with the new rating.
  
### **Expected Behavior**
All buttons (Like, Book, Accept, Complete, Rate) should update Firebase in real-time.
The UI should reflect the updated values dynamically.

### **Additional Information**
**Tech Stack:** Next.js, TypeScript, Tailwind CSS, Firebase
**Firebase Features:** Firestore for data storage, Authentication for user management

### **Deadline**
I aim to have this functionality implemented and fully tested by January 25, 2025.

### **Acceptance Criteria**
The "Like" button increments the like count in Firebase and updates the UI.
The "Book" button creates a booking request in Firebase with all necessary details.
Workers can accept or complete bookings via the dashboard, and the status updates in Firebase.
The "Rate" button works as expected, storing and displaying ratings.

**Comment by Lut-Lat-Aung** [comment]
I would like to help you out with the Like button.

**Comment by Siddesh0002T** [comment]
@Lut-Lat-Aung Thank you for offering your assistance with the Like button; I truly appreciate it. If you find my work valuable, I kindly request you to star my repository and follow me on GitHub. Your support is greatly appreciated!

**Comment by Nithesh0812** [comment]
Hey I'd like to help you out with your issue.

**Comment by Siddesh0002T** [comment]
@Nithesh0812  Yes, I'd appreciate your help! To give you access to Firebase, could you please provide me with the following details:
Your Firebase email address
The role you'd like to have (e.g., Editor, Viewer)
Once I have that, I can add you to the Firebase project.

**Comment by Nithesh0812** [comment]
@Siddesh0002T 
my email - mudsunithesh0812@gmail.com
role - Editor

**Comment by Nithesh0812** [comment]
Hey, Can you provide me access to your firebase

**Comment by Siddesh0002T** [comment]
@Nithesh0812  I have granted editor access to mudsunithesh0812@gmail.com in Firebase. You can proceed with your work on the project.

**Comment by Siddesh0002T** [comment]
@Nithesh0812  Could you please complete it at your earliest convenience?

**Comment by Siddesh0002T** [comment]
Thank you @Lut-Lat-Aung  for your contribution! The "Like" feature works well, and I appreciate the implementation. Users can now like the professional, and the like count updates correctly. Additionally, the functionality to like once and undo the like is a great touch.

I’ve reviewed the changes, and I’m accepting your pull request. However, there is still one issue to address: after the page is refreshed, the state of the "Like" button resets. It would be great if you could implement a persistent state for the "Like" button, so that after refreshing the page, the button reflects the correct state (liked or not liked).

Once this is addressed, the feature will work seamlessly. I look forward to your updated pull request!

Thanks again for your hard work!

**Comment by Siddesh0002T** [comment]
However, the rating and booking features still need to be addressed. If you could focus on resolving those issues, it would be great. Once those are fixed, the project will be fully functional.

Keep up the great work, and feel free to reach out if you need any further assistance or clarification.

Looking forward to your next update!

**Comment by Lut-Lat-Aung** [comment]
@Siddesh0002T 
I am glad that my contribution is helpful. I will try to fix the persistent state for "Like" button soon. However, for booking and rate features, if I have free time after midterm, I will look into those as well. I have midterm exams coming in January and the exams finish at the last week of January. Thanks for your understanding.

**Comment by Siddesh0002T** [comment]
@Lut-Lat-Aung  Thank you for your efforts and contributions! Your work on fixing the persistent state for the 'Like' button and considering the booking and rate features is greatly appreciated. Best of luck with your midterm exams in January — I hope they go smoothly and successfully. Take your time, and we'll look forward to any updates whenever you're ready. Thanks again for your dedication!

**Comment by Nithesh0812** [comment]
@Siddesh0002T I have tried to rectify the bug where on refreshing the page the state of the Like Button resets.

**Comment by Siddesh0002T** [comment]
@Nithesh0812 If the bug is fixed, please send a pull request so we can review and merge it into the main branch.

**Comment by Siddesh0002T** [comment]
Thank you, @Nithesh0812, for fixing the "Like" button issue! The persistent state is working perfectly now, and the feature is seamless.

Next, please focus on resolving the issues with the "Book" button and the "Rating" button. Let’s aim to get those functionalities working smoothly as soon as possible. Let me know if you need any additional details or support.

Great work, and keep it up!

**Comment by Siddesh0002T** [comment]
I've fixed the rating button. Now, only the booking system remains to be fixed. I'll address it as soon as possible!
